bokehjs/src/coffee/polyfill.ts

```diff
@@ -0,0 +1,9 @@
+declare function require(moduleName: string): any
+
+if (typeof WeakMap !== "function") {
+  require("es6-weak-map/implement")
+}
+
+if (typeof Set !== "function") {
+  require("es6-set/implement")
+}
```

Comment:
@bryevdv, phantomjs support is actually the easy case, because we control the environment, at least as far testing infrastructure and static image generation code (which is based on it) go, because we can always inject whatever code we need prior to running bokehjs and client code. IE is harder, because we don't know where bokehjs will actually run. However, I just recalled there is this MS' proprietary conditional syntax in HTML comments, that can be used for conditionally including polyfills. I will go ahead and implement this in another PR (when this is merged).